### DISCUSSION TITLE
Add Plex Label based on user requesting

### DISCUSSION THREAD

**Discussion by bmpreston** [sct/overseerr#2940] [Feature Requests]
I would love to see the option to auto add Label to the media in Plex based on the user requesting.  

Scenario:  (This is real world for me)

Wife uses Overseerr to add media, a TV show.  

1. TV show is approved
2. TV show is added into Sonarr 
3. Sonarr processes, adds and does appropriate actions within SAB, etc etc
4. Sonarr imports properly receive files
5. Plex scans library and adds to library
**6.  Plex is instructed by Overseerr to add label to library as library was requested by user:  Wife**
7.  Because of filtering in Plex Home, account _WIFE_ gains access to media in Plex.  

I would also see a case to make 'existing media' add the label onto the Plex library.  This would ensure that if Media _ALREADY_ exists, it would become available in Plex to the user.   This might mean you need to know previously during a Plex scan if the library exists, and the user accounts filters.  

In case there is any discussion or question on why I do this, it's so my wife only has to see her 10+ shows, not ALL shows on the server. 

Amazing product!  Thank you for doing this!

![Screen Shot 2022-08-17 at 22 48 51](https://user-images.githubusercontent.com/29456939/185282578-833f1a8c-70ba-4b07-8f9b-93b9eaf74c71.png)
![Screen Shot 2022-08-17 at 22 51 36](https://user-images.githubusercontent.com/29456939/185282580-d5636b96-cf69-4822-9db4-a45961c56cf3.png)

**Comment by RichBrew** [comment]
I use something like this too, by only sharing items with the 'public' label to users outside of my home. It would be good for the same tags to also be applied to Sonarr/Radarr as well. This way, recently added, requested, coming soon, could also be filtered in Overseerr to only include items with labels that have been attributed to the logged in user.

**Comment by Nitrousoxide** [comment]
This would also be useful to me as you can split up what shows are visible to who based on who requested them. This would require some more work on Overseerr's end though too I'd think.  Since you probably wouldn't want overseerr to be displaying a show as downloaded and available when a different user other than the one logged in requested it.  If you set a tag to be associated with a user you'd want to only show the videos with that tag in plex to the user in overseerr.

It might also require some way to handle two users requesting the same show/movie.  I presume it'd just give the same video both tags on plex, so then would become available to both users.  Obviously you don't want to waste space downloading the same video twice.

**Comment by c11umw** [comment]
There is a way to do this, but it takes a little bit of setting up.
Assuming Plex libraries are in /media 

Plex:
TV folder: /media/tv
Wife's programs: /media/tv/wifey
Plex TV library "Add folders to your library" add **both** the libraries above so TV has all content
Plex TV_Wifey library "Add folders to your library" only add /media/tv/wifey

TV library needs to have both paths explicitly defined as it doesn't reliably propagate down a nested path. Adding both resolves it.
I found that one out the hard way when content started disappearing.

Sonarr:
Docker settings (if CLI)
```
-v /media/tv:/TV \
-v /media/tv/wifey:/TV_Wifey \
```
Then in the application
Settings | media management: Root Folders - add both TV and TV_Wifey as root folders
No need to {re}do Series | Library Import unless there's existing content you're monitoring.

Overseer:
Once it's synced with Sonarr then both TV and TV_Wifey Root Folders will be options when requesting content.
Wifey just needs to select root folder: TV_Wifey when requesting and Sonarr will file it to the right folder.
If the TV folder is requested accidentally (or there are existing requests) then it can be manually edited in Sonarr to correct the path.
I think the user's permissions just needs to have "Advanced Requests" selected to enable the root folder option when requesting, otherwise the user downloads to the default path.

I do something similar to the above with a "Temp" folder for content that's not worth buying/keeping. I setup an purge process that auto-removes films and series from their temp folders after 60-90 days. Keeps our library clean and they have enough time to watch their cheesy content

**Comment by lordhelmet77** [comment]
IMO this should be implemented in Sonarr/Radarr rather than Overseer:

Overseer's purpose is to discover and request media, with download and file handling through Sonarr/Radarr. Overseer can already set a tag for the person requesting (Service Settings > Tag Requests) that is carried over to Sonarr/Radarr, but those apps have no means for tags to carry over to Plex.
